.github/workflows/build.yml

```
@@ -15,7 +15,15 @@ permissions:
   contents: read
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}-reusable
+  # For PRs, this names the group after the workflow ('Tests'), 
+  # the PR number followed by '/merge', and the name of the branch being merged
+  #   example: Tests-123456/merge-mycoolbranch
+  #
+  # For anything else that triggers this workflow, it names the group after
+  # the workflow ('Tests'), the branch, and the sha hash
+  #   example: Tests-main-(40 char sha hash)
+  #
+  group: ${{ github.workflow }}-${{ github.ref_name }}-${{ github.head_ref || github.sha }}
```

Remove excess verbosity; simplify (only add 'actor_id')
```suggestion
  # https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#concurrency
  # 'group' must be a key uniquely representing a PR or push event.
  # github.workflow is the workflow name
  # github.actor_id is the user invoking the workflow
  # github.head_ref is the source branch of the PR or otherwise blank
  # github.run_id is a unique number for the current run
  group: ${{ github.workflow }}-${{ github.actor_id }}-${{ github.head_ref || github.run_id }}
```

---

Plz don't. It'll defeat the entire purpose of the cancelation scopes..

---

@webknjaz can you elaborate? The suggestion here is what we use at present, dismbiguated with the username. Unsure how this 'defeat[s] the entire purpose'.

---

Shall we use `github.actor` rather than `github.actor_id`? The username is more understandable when needing to read/debug what's going on.

(Theoretically usernames can change, but in practice is unlikely to be of concern.)

---

Sure, I've edited the suggestion. 

A

---

@AA-Turner I forgot to post but we talked this through w/ Kira and concluded that your suggestion was reasonable and I was overthinking it. I originally thought that the fact that it wouldn't produce the same group name for pushes in upstream was undesired but realized later that it's fine.